packages/mds-types/transformers/vehicle_event_transformers.ts

```diff
@@ -0,0 +1,271 @@
+import {
+  VEHICLE_EVENT,
+  VehicleEvent_v0_4_1,
+  VEHICLE_REASON_0_4_1,
+  VEHICLE_EVENT_0_4_1,
+  VehicleEvent,
+  VEHICLE_STATE
+} from '../index'
+
+function convert_event_type_and_event_type_reason_to_event_type(
+  event_type: VEHICLE_EVENT_0_4_1,
+  event_type_reason: VEHICLE_REASON_0_4_1 | null | undefined
+): VEHICLE_EVENT {
+  switch (event_type) {
+    /* Included for the sake of completeness. No provider will ever submit such
+     * an event type through the agency API.
+     */
+    case 'agency_drop_off': {
+      return 'agency_drop_off'
+    }
+    case 'agency_pick_up': {
+      return 'agency_pick_up'
+    }
+    case 'cancel_reservation': {
+      return 'reservation_cancel'
+    }
+    case 'deregister': {
+      switch (event_type_reason) {
+        case 'decommissioned': {
+          return 'decommissioned'
+        }
+        case 'missing': {
+          return 'missing'
+        }
+        default: {
+          return 'unspecified'
+        }
+      }
+    }
+    case 'provider_drop_off': {
+      return 'provider_drop_off'
+    }
+    case 'provider_pick_up': {
+      switch (event_type_reason) {
+        case 'rebalance': {
+          return 'rebalance_pick_up'
+        }
+        case 'maintenance': {
+          return 'maintenance_pick_up'
+        }
+        case 'charge': {
+          return 'maintenance_pick_up'
+        }
+        case 'compliance': {
+          return 'compliance_pick_up'
+        }
+        default: {
+          return 'unspecified'
+        }
+      }
+    }
+    case 'reserve': {
+      return 'reservation_start'
+    }
+    case 'service_start': {
+      return 'on_hours'
+    }
+    case 'service_end': {
+      switch (event_type_reason) {
+        case 'low_battery': {
+          return 'battery_low'
+        }
+        case 'maintenance': {
+          return 'maintenance'
+        }
+        case 'compliance': {
+          return 'compliance_pick_up'
+        }
+        case 'off_hours': {
+          return 'off_hours'
+        }
+        default: {
+          return 'unspecified'
+        }
+      }
+    }
+    case 'trip_end': {
+      return 'trip_end'
+    }
+    case 'trip_enter': {
+      return 'trip_enter_jurisdiction'
+    }
+    case 'trip_leave': {
+      return 'trip_leave_jurisdiction'
+    }
+    case 'trip_start': {
+      return 'trip_start'
+    }
+    default: {
+      return 'unspecified'
+    }
+  }
+}
+
+export function convert_event_type_to_vehicle_state(event_type: VEHICLE_EVENT_0_4_1): VEHICLE_STATE {
```

Comment:
You could probably have this be the same function as the one that handles event_type & event_type_reason and avoid having to redefine the whole switch statement again.

Reply:
I don't see how that would work, especially since the conversions aren't 1:1.